module/VuFind/src/VuFind/Controller/AbstractBase.php

```diff
@@ -736,16 +735,45 @@ protected function normalizeUrlForComparison($url)
         return trim(end($parts), '/');
     }
 
+    /**
+     * Checks if a followup url is set
+     *
+     * @return bool
+     */
+    protected function hasFollowupUrl()
+    {
+        return null !== $this->followup()->retrieve('url');
+    }
+
     /**
      * Retrieve a referer to keep post-login redirect pointing
      * to an appropriate location.
      * Unset the followup before returning.
      *
+     * @param bool $checkRedirect Whether the query should be checked for param 'redirect'
+     *
      * @return string
      */
-    protected function getFollowupUrl()
-    {
-        return $this->followup()->retrieve('url', '');
+    protected function getAndClearFollowupUrl($checkRedirect = false)
+    {
+        if ($url = $this->followup()->retrieve('url', '')) {
+            $lightboxParent = $this->followup()->retrieveAndClear('lightboxParent');
+            $this->clearFollowupUrl();
```

Comment:
Is there a strong reason to clear this separately rather than using retrieveAndClear above? If so, is it accurate to call the method getAndClearFollowupUrl if it's only clearing conditionally?

Reply:
You are right. Now retrieveAndClear is used and clearFollowupUrl is removed.